modules/file_writer_module/tests/test_file_writer_module.cpp

```diff
@@ -0,0 +1,514 @@
+    //ASSERT_EQ(fb.getPropertyValue("DownScalingDevider"), 5);
+
+    // Try to set negative batch cycle
+    fb.setPropertyValue("BatchCylce", -1);
+    ASSERT_EQ(fb.getPropertyValue("BatchCylce"), 1);
+
+    // Try to set min value
+    fb.setPropertyValue("BatchCylce", 0);
+    ASSERT_EQ(fb.getPropertyValue("BatchCylce"), 1);
+
+    // Try to set max value
+    fb.setPropertyValue("BatchCylce", 3601);
+    ASSERT_EQ(fb.getPropertyValue("BatchCylce"), 3600);
+}
+
+ 
+TEST_F(FileWriterModuleTest, SchemaTest)
+{
+    auto ctx = CreateContext();
+    const auto module = CreateModule();
+
+    auto fb = module.createFunctionBlock("FileWriterModuleParquet", nullptr, "id");
+    ASSERT_TRUE(fb.assigned());
+
+    createSignals(ctx);
+    fb.getInputPorts()[0].connect(doubleSignal);
+    fb.getInputPorts()[1].connect(floatSignal);
+    fb.getInputPorts()[2].connect(int32Signal);
+    fb.getInputPorts()[3].connect(int64Signal);
+    //fb.getInputPorts()[4].connect(int16Signal);
+    //fb.getInputPorts()[5].connect(int8Signal);
+
+    // Execute recording
+    fb.setPropertyValue("FileName", "Schema");
+    fb.setPropertyValue("RecordingActive", true);
+    fb.setPropertyValue("BatchCylce", 3600);
+
+    // Send Data
+    sendSignalWithTime1<daq::SampleType::Float64>(1, 50, 1000);
+    sendSignalWithTime1<daq::SampleType::Float32>(1, 50, 1000, 0, false);
+    sendSignalWithTime1<daq::SampleType::Int32>(1, 50, 1000, 0, false);
+    sendSignalWithTime1<daq::SampleType::Int64>(1, 50, 1000, 0, false);
+    //sendSignalWithTime1<daq::SampleType::Int16>(1, 50, 1000, 0, false);
+    //sendSignalWithTime1<daq::SampleType::Int8>(1, 50, 1000, 0, false);
+
+    // Recording to false triggers file write
+    fb.setPropertyValue("RecordingActive", false);
+
+    // Created Recorded files have a file format
+    // <fileName>_x_y.parquet
+    // where x is table where signals are sharing a domain signal
+    // where y is subcount if file batching is done.    
+    arrow::MemoryPool* pool = arrow::default_memory_pool();
+    std::shared_ptr<arrow::io::RandomAccessFile> input;
+    PARQUET_ASSIGN_OR_THROW(input, arrow::io::ReadableFile::Open("Schema_1_0.parquet"));
+
+    // Open Parquet file reader
+    std::unique_ptr<parquet::arrow::FileReader> arrow_reader;
+    PARQUET_THROW_NOT_OK(parquet::arrow::OpenFile(input, pool, &arrow_reader));
+
+    // Read entire file as a single Arrow table
+    std::shared_ptr<arrow::Table> table;
+    PARQUET_THROW_NOT_OK(arrow_reader->ReadTable(&table));
+
+    ASSERT_EQ(table->schema()->field(0)->type().get()->name(), "int64");
+    ASSERT_EQ(table->schema()->field(1)->type().get()->name(), "double");
+    ASSERT_EQ(table->schema()->field(2)->type().get()->name(), "float");
+    ASSERT_EQ(table->schema()->field(3)->type().get()->name(), "int32");
+    ASSERT_EQ(table->schema()->field(4)->type().get()->name(), "int64");
+    //ASSERT_EQ(table->schema()->field(5)->type().get()->name(), "int16");
```

Comment:
ToDo: Strange behaviour from Arrow: Alone they are working - if a 5th one is added by writing the file an error occurs in the arrow lib.